java-checks/src/main/java/org/sonar/java/checks/InnerStaticClassesCheck.java

```diff
@@ -69,13 +69,22 @@ public void visitClass(ClassTree tree) {
     scan(tree.members());
     Boolean oneReference = atLeastOneReference.pop();
     outerClasses.pop();
-    if (!symbol.isStatic() && !oneReference && !outerClasses.isEmpty() && isFirstParentStatic(outerClasses)) {
+    if (!symbol.isStatic() && !oneReference && hasOneStaticParent(outerClasses) && !hasNonStaticSuperClass(tree)) {
       String named = symbol.name().isEmpty() ? "named " : "";
       context.addIssue(tree, this, "Make this a " + named + "\"static\" inner class.");
     }
   }
 
-  private static boolean isFirstParentStatic(Deque<Symbol> outerClasses) {
+  private boolean hasNonStaticSuperClass(ClassTree tree) {
+    if (tree.superClass() != null) {
+      Type superClass = tree.symbol().superClass();
+      final Symbol superClassSymbol = superClass.symbol();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'superClass' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)